experimental/AlgebraicShifting/src/PartialShiftGraph.jl

```diff
@@ -0,0 +1,309 @@
+Abstract simplicial complex of dimension 1 on 4 vertices
+
+julia> shifts = partial_shift_graph_vertices(QQ, K, weyl_group(:A, 3))
+6-element Vector{SimplicialComplex}:
+ Abstract simplicial complex of dimension 1 on 4 vertices
+ Abstract simplicial complex of dimension 1 on 4 vertices
+ Abstract simplicial complex of dimension 1 on 4 vertices
+ Abstract simplicial complex of dimension 1 on 4 vertices
+ Abstract simplicial complex of dimension 1 on 4 vertices
+ Abstract simplicial complex of dimension 1 on 4 vertices
+
+julia> facets.(shifts)
+6-element Vector{Vector{Set{Int64}}}:
+ [Set([3, 1]), Set([4, 1]), Set([2, 1])]
+ [Set([3, 1]), Set([2, 3]), Set([2, 1]), Set([4])]
+ [Set([3, 1]), Set([4, 2]), Set([2, 1])]
+ [Set([4, 3]), Set([3, 1]), Set([2, 1])]
+ [Set([2, 1]), Set([2, 3]), Set([4, 2])]
+ [Set([2, 1]), Set([2, 3]), Set([4, 3])]
+```
+"""
+function partial_shift_graph_vertices(F::Field,
+                                      K::SimplicialComplex,
+                                      W::Union{WeylGroup, Vector{WeylGroupElem}})
+  current = K
+  visited = [current]
+  phi = isomorphism(PermGroup, parent(first(W)))
+  # by properties of algebraic shifting
+  # we know that K will be the last in this sorted list
+  # sorting here should also speed up unique according to julia docs
+  unvisited = unique(
+    x -> Set(facets(x)),
+    sort([exterior_shift(F, K, phi(w)) for w in W]; lt=isless_lex))[1:end - 1]
+
+  while !isempty(unvisited)
+    current = pop!(unvisited)
+    push!(visited, current)
+    shifts = unique(
+      x -> Set(facets(x)),
+      sort([exterior_shift(F, current, phi(w)) for w in W]; lt=isless_lex))[1:end - 1]
+
+    # dont visit things twice
+    new_facets = filter(x -> !(x in Set.(facets.(visited))), Set.(facets.(shifts)))
+    unvisited = simplicial_complex.(
+      union(Set.(facets.(unvisited)), new_facets))
+  end
+  return sort(collect(visited); lt=isless_lex)
+end
+
+""" Compute the multi edges, that is, for each complex `K`  compute which
+    other complexes can be reached by applying the partial shifts `deltas`
+    to K, and store the shift matrices that give rise to each edge."""
+function multi_edges(F::Field,
+                     permutations::Vector{PermGroupElem},
+                     complexes::Vector{Tuple{Int,T}},
+                     complex_labels::Dict{Set{Set{Int}}, Int}
+) where T <: ComplexOrHypergraph;
+  # For each complex K with index i, compute the shifted complex delta of K by w for each w in W.
+  # For nontrivial delta, place (i, delta) → w in a singleton dictionary, and eventually merge all dictionaries
+  # to obtain a dictionary (i, delta) → [w that yield the shift K → delta]
+  return reduce(
+  (d1, d2) -> mergewith!(vcat, d1, d2), (
+    Dict((i, complex_labels[Set(facets(delta))]) => [p])
+    for (i, K) in complexes
+      for (p, delta) in ((p, exterior_shift(F, K, p)) for p in permutations)
+        if !issetequal(Set(facets(delta)), Set(facets(K))));
```

Comment:
```suggestion
        if !issetequal(facets(delta), facets(K)));
```
issetequals turns the input into sets itself, if it finds it sensible